Implement `siunit_length` mapcss function

Famlam opened the pull request:
Implements the new `siunit_length` mapcss function.

Rules using this function were [added in the JOSM mapcss rules today](https://josm.openstreetmap.de/changeset?old_path=%2Fjosm&old=19129&new_path=%2Fjosm&new=19129).
(If desired I can update the mapcss rules in this PR)

I'm using Pint for the unit conversions. Pint would also be useful for #2251 (as suggested by @jocelynj) and probably other places as well (plugin Number.py for example).

frodrigo commented on plugins/tests/test_mapcss_parsing_evaluation.validator.mapcss:
```diff
@@ -561,3 +561,27 @@ node[to_float("3.14e1") = 31.4][to_double("3.1415e1") = 31.415] {
+  assertMatch: "node tag_length=\"10ft 1in\" b=3.0734";
+  assertMatch: "node tag_length=\"20'2\"\"   b=6.1468";
+  assertMatch: "node tag_length=\"-20'2\"\"  b=-6.1468";
+  assertNoMatch: "node tag_length=\"Hello\"  b=0";
```
Should really be 0 here ?

Famlam replied:
No, it's just to test that it doesn't error, but there shouldn't be any match (hence the assertNoMatch). I can also add a test like this if you prefer:
`node[any(siunit_length("Hello"), "x") == "x"]`
This would literally ensure the output is None.

frodrigo replied:
Ok, it is NoMatch, so it is ok for me.

Famlam commented:
I spotted two different (minor) bugs:
1. JOSM always converts to Double, I didn't, so our output differed from JOSM by a `.0` for an int -> int conversion
2. In the test string I was still using `tag(a)` while I had renamed that later to `tag_length`

Also removed a lot of unnecessary `\"`

Famlam commented:
Hmm, I just discovered that pint also does mathematics... E.g. input `4-5` (a range) becomes -1 rather than PintError...
This complicates it's use quite a bit...

Famlam commented:
New implementation in #2263
